todo_app/data/storage_trello.py

```diff
@@ -0,0 +1,117 @@
+""" Persistence layer, trello edition
+"""
+import os 
+
+from flask import session
+import dotenv
+import requests
+
+from .TrelloSession import TrelloSession
+from .. import trello_config
+
+dotenv.load_dotenv()
+
+trello = TrelloSession('https://api.trello.com', 
+    os.getenv('TRELLO_API_KEY'), 
+    os.getenv('TRELLO_TOKEN') )
+
+VALID_STATUSES=trello_config.LISTS.keys()
+
+class Card(object):
+    """ 
+    Sort-of POD for a todo card, but coupled to trello because of its 
+    conversion methods
+    """
+    def __init__(self, id, title, status):
+        self.id = id
+        self.title = title
+        if status not in VALID_STATUSES:
+            raise ValueError(f'Attempt to create a card with status "{status}", which was not in {VALIDstatusES}.')
+        self.status = status
+
+    def to_trello(self):
+        """ Return JSON suitable for a POST or PUT to trello """
+        id_of_list = trello_config.LISTS[self.status]
+        
+        return { 'id': self.id,
+                'idList': id_of_list,
+                'name': self.title }
+
+    @classmethod
+    def from_trello(cls, trello_card):        
+        """ Factory to create Card from the trello JSON for a card """
+        # Look up the key of the dict given its value. Behavior is not defined 
+        # if  it's not present.
+        for status in trello_config.LISTS.keys():
+            if trello_config.LISTS[status] == trello_card['idList']:
+                break
+
+        return cls(trello_card['id'], trello_card['name'], status)
+
+
+def get_items():
+    """
+    Fetches all saved items from trello
+
+    Returns:
+        list: The list of saved items.
+    """
+    results = []
+
+    board_id = trello_config.BOARD_ID
+
+    url = trello.request_url(f'/1/board/{board_id}/lists/', {'cards': 'open'} )
+    data = trello.retrieve_json(url)
```

Comment:
Rather than pre-encoding the parameters in the URL, we could let `requests` handle that for us like:
```
requests.get(url, params = query_dict)
```

So here, we could let retrieve_json handle the merging of the standard & request-specific query parameters, and just have:
```suggestion
    url = trello.request_url(f'/1/board/{board_id}/lists/')
    data = trello.retrieve_json(url, params = {'cards': 'open'} )
```

Reply:
I like that it's using library code rather than needing a supporting function, but I dislike that the thing coming back from trello.request_url is not the whole url. time for some more head scratching, thanks.

Reply:
I like it, I've raised it as an issue in this repo because I just tried to do this and **kwargs at the same time, and found out I'm too tired.